packages/icons-vue/tasks/build.js

```diff
@@ -5,13 +5,216 @@
+      // Special case here, we need to coordinate that we are using title,
+      // potentially, to get the right focus attributes
+      title: props.title,
+      ...data.attrs,
+    });
+    const svgData = {
+      attrs,
+      on: listeners,
+    };
+
+    if (data.staticClass) {
+      svgData.class = {
+        [data.staticClass]: true,
+      };
+    }
+
+    if (data.class) {
+      svgData.class[data.class] = true;
+    }
+
+    return createElement('svg', svgData, [
+      props.title && createElement('title', null, props.title),
+      ${content.map(toString).join(', ')},
+      children,
+    ]);
+  },
+};`;
+}
+
+function toString(descriptor) {
+  const { elem, attrs = {}, content = [] } = descriptor;
+  if (content.length === 0) {
+    return `createElement('${elem}', { attrs: ${JSON.stringify(attrs)} })`;
+  }
+  return `createElement('${elem}', { attrs: ${JSON.stringify(
+    attrs
+  )} }, [${content.map(toString).join(', ')}])`;
+}
+
+function createIconStory(info) {
+  const { moduleName, outputOptions } = info;
+  return `import { storiesOf } from '@storybook/vue';
+import { action } from '@storybook/addon-actions';
+import ${moduleName} from '../../../${outputOptions.file}';
+
+storiesOf('${moduleName}', module)
+  .add('default', () => ({
+    components: {
+      icon: ${moduleName},
+    },
+    template: \`<icon></icon>\`,
+  }))
+  .add('with title', () => ({
+    components: {
+      icon: ${moduleName},
+    },
+    template: \`<icon><title slot="title">${moduleName}</title></icon>\`,
+  }))
+  .add('with aria-label and tabindex', () => ({
+    components: {
+      icon: ${moduleName},
+    },
+    template: \`<icon aria-label="${moduleName}" tabindex="0"></icon>\`,
+  }))
+  .add('with event handler', () => ({
+    components: {
+      icon: ${moduleName},
+    },
+    template: \`<icon v-on:click="action"></icon>\`,
+    methods: {
+      action: action('clicked'),
+    },
+  }))
+  .add('with custom classes', () => ({
+    components: {
+      icon: ${moduleName},
+    },
+    template: \`<icon class="custom classes" :class="'dynamic classes_2'"></icon>\`,
+  }));`;
+}
+
+const STORYBOOK_DIR = path.resolve(__dirname, '../examples/storybook/stories');
+
+async function build({ cwd }) {
+  reporter.info(`Building components for ${meta.length} icons...`);
+  const ESM_DIR = path.join(cwd, 'es');
+  const BUNDLE_FORMATS = [
+    {
+      format: 'cjs',
+      directory: 'lib',
+    },
+    {
+      format: 'umd',
+      directory: 'umd',
+    },
+  ];
+
+  reporter.info('Building ESM and bundle sources...');
+  await Promise.all(
+    meta.map(async info => {
+      const source = createModuleFromInfo(info);
+      const jsFilepath = path.join(cwd, info.outputOptions.file);
+
+      await fs.ensureDir(path.dirname(jsFilepath));
+      await fs.writeFile(jsFilepath, source);
+
+      await Promise.all(
+        BUNDLE_FORMATS.map(async ({ format, directory }) => {
+          const bundle = await rollup({
```

Comment:
Would you be interested in refactoring this bit so that we're not creating a new rollup bundle instance for each file?

Reply:
Absolutely!

Reply:
Let me know if you want any help with this, for the icons-react side-of-things we took advantage of rollup's codesplitting features to accomplish this: https://github.com/carbon-design-system/carbon/blob/master/packages/icons-react/tasks/build.js#L177

Which is basically that the `input` field can be an object of `outputName: filename` options, and similarly `filename` could be a virtual filename provided by the `virtual` plugin. The `outputName` is then used to figure out what directory to place things in, for example: `'add--filled/16': 'virtual_add_filled_16'` would output to `lib/add--filled/16` in the CommonJS step. (not saying these things should be named this way, just using as an example!)